hr_employee_calendar_planning/models/resource_resource.py

```diff
@@ -0,0 +1,10 @@
+# Copyright 2019 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResourceResource(models.Model):
+    _inherit = 'resource.resource'
+
+    calendar_id = fields.Many2one(required=False)
```

Comment:
Why are you changing this as not required?

Reply:
You need it in order to be able to execute employee.resource_calendar_id = False at the end of the post init hook since it is a related field linked to calendar_id, which in 11.0 is required.

Reply:
That was for regenerating the calendar, but I added at the end a check also on active field, so you can deactivate the calendar instead of this.

Reply:
Okay, so can i just delete employee.resource_calendar_id = False and leave calendar_id as it was?

Reply:
You have to replace it by `employee.resource_calendar_id.active = False`

Reply:
Doing so causes two tests to fail because we dont enter this if in regenerate_calendar:
```
        if not self.resource_calendar_id or self.resource_calendar_id.active:
            self.resource_calendar_id = self.env['resource.calendar'].create({
                'active': False,
                'name': _(
                    'Auto generated calendar for employee'
                ) + ' %s' % self.name,
                'attendance_ids': [],
            }).id
```

I think that removing `employee.resource_calendar_id = False` and keeping `active` equal to True is what we need and makes all tests pass.

Reply:
Right, I have looked incorrectly at the regenerate method. Please make sure that after this, you switch again the calendar to active=False.

Reply:
Yes, on the creation dictionary there is already `'active': False`.

Reply:
Yeah, but that are the split ones. I'm talking about the original one.

Reply:
okay